### PR TITLE
[DOC-11930] Move Eventing nav to docs-devex

### PR DESCRIPTION
**Pull request by julia-cb**
Changes in this PR:

- New file with Eventing nav
- Updated overview page title to "Run a Function on Data Change"

docs-server PR: https://github.com/couchbase/docs-server/pull/3525
preview site: https://preview.docs-test.couchbase.com/eventnav/server/current/eventing/eventing-overview.html

### CODE REVIEW

File: modules/eventing/partials/nav.adoc
Comment (sarahlwelton) [inline modules/eventing/partials/nav.adoc:1]: Are all of these files on docs-devex now, or do the files still live on docs-server?

**Review by sarahlwelton — APPROVED** [review]
Seems straightforward to me!

File: modules/eventing/partials/nav.adoc
Comment (julia-cb) [reply modules/eventing/partials/nav.adoc:1]: All files are in docs-devex!

**Review by simon-dew — COMMENTED** [review]
LGTM 👍

**Pull request merged**